roofit/roofitcore/src/TestStatistics/RooSumL.cxx

```diff
@@ -0,0 +1,90 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#include <TestStatistics/RooSumL.h>
+#include <RooAbsData.h>
+#include <TestStatistics/RooSubsidiaryL.h>
+
+#include <algorithm> // min, max
+
+namespace RooFit {
+namespace TestStatistics {
+
+// Note: components must be passed with std::move, otherwise it cannot be moved into the RooSumL because of the unique_ptr!
+RooSumL::RooSumL(RooAbsPdf *pdf, RooAbsData *data, std::vector<std::unique_ptr<RooAbsL>> components, RooAbsL::Extended extended)
+   : RooAbsL(pdf, data,
+             data->numEntries(), // TODO: this may be misleading, because components in reality will have their own N_events...
+             components.size(), extended), components_(std::move(components))
+{}
+
+
+bool RooSumL::processEmptyDataSets() const
+{
+   // TODO: check whether this is correct! This is copied the implementation of the RooNLLVar override; the
+   // implementation in RooAbsTestStatistic always returns true
```

Comment:
Did somebody check? I'm seeing too many TODOs in this PR.

Reply:
Replaced with a note to possible future implementers of alternative test statistic types (e.g. Chi^2).

Reply:
No, actually, this function is no longer used, its functionality was moved to likelihood_builders, so put the comment there in the appropriate place and removed this function.